.ci/generate_documentation

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+import os
+import pathlib
+import subprocess
+import tempfile
+
+import git
+
+import gitutil
+import github.util
+from util import (
+    check_env,
+    ctx,
+    fail,
+    which,
+)
+
+# Verify that required env_vars are set
+sphinx_executable = which('sphinx-build')
+source_repo_dir = pathlib.Path(check_env('SOURCE_PATH')).resolve()
+source_relative_doc_root_dir = check_env('REL_DOC_ROOT_DIR')
+
+# Verify optional env_vars
+# In anticipation of changes to pipeline contract, use '<logical_name>_COMPONENT_NAME>
+# (e.g.: 'github.com/gardener/cc-utils')
+# TODO: rm comment once changes are made
+target_component_name = os.environ.get('TARGET_COMPONENT_NAME', None)
```

Comment:
immediately make this a hard requirement (util.check_env). use step: <step_name>.vars to pass env vars